src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -169,15 +169,14 @@ protected function refreshUser(TokenInterface $token)
 
             try {
                 $refreshedUser = $provider->refreshUser($user);
-                $token->setUser($refreshedUser);
+                $newToken = clone $token;
+                $newToken->setUser($refreshedUser);
```

Comment:
actually couldn't we avoid cloning and keep using the original token? The user stored in the token would always be the last loaded user here, I see no possible side effect, do I miss something?

Reply:
The problem is exactly that the user stored in the token would always be the last user found. So the user in the token changes, we shouldn't compare the last found user with the user in the token but with the original user, otherwise the token will stay deauthenticated.

Reply:
indeed 👍